queue/src/fiskaltrust.Middleware.Localization.QueueDE/Repositories/ReadOnlyReceiptReferenceRepository.cs

```diff
@@ -102,46 +97,21 @@ private async Task<bool> AddReference(HashSet<ReceiptReferencesGroupedData> rece
             var responseTarget = JsonConvert.DeserializeObject<ReceiptResponse>(target.response);
             var responseSource = JsonConvert.DeserializeObject<ReceiptResponse>(source.response);
 
-            (var znrSource, _) = await GetLastZNumberForQueueItem(source).ConfigureAwait(false);
+            var dailyClosingSource = dailyClosings.Where(x => x.ZTime >= source.cbReceiptMoment).FirstOrDefault();
 
             var receiptReference = new ReceiptReferencesGroupedData()
             {
-                RefMoment = source.cbReceiptMoment,
+                RefMoment = dailyClosingSource.ZTime,
```

Comment:
We should rename `RefMoment` to `TargetZMoment` to make it clear internally what this moment represents.

Do we know if its the same in the DFKA? The `ReceiptReferenceData` is also used there.